core/src/core/block.rs

```diff
@@ -212,9 +211,15 @@ impl Readable for Block {
 		let (input_len, output_len, proof_len) =
 			ser_multiread!(reader, read_u64, read_u64, read_u64);
 
-		let inputs = try!((0..input_len).map(|_| Input::read(reader)).collect());
-		let outputs = try!((0..output_len).map(|_| Output::read(reader)).collect());
-		let kernels = try!((0..proof_len).map(|_| TxKernel::read(reader)).collect());
+		let inputs: Vec<_> = try!((0..input_len).map(|_| Input::read(reader)).collect());
```

Comment:
We could probably take this a step further and wrap this up in something like `read_sorted` to match `write_sorted` above.